### PR TITLE
Make storageClassName configurable

### PR DESCRIPTION
**Pull request by etichy**

### CODE REVIEW

File: deploy/helm/values.schema.json [hunk 1]
```diff
@@ -250,6 +250,13 @@
                 "type": "string"
             }
         },
+        "storageClassName": {
+            "description": "Kubernetes storage classes. More info: https://kubernetes.io/docs/concepts/storage/storage-classes",
+            "type": "object",
+            "additionalProperties": {
+               "type": "string"
+            }
+        },
```
Comment (pstranak-sw) [inline deploy/helm/values.schema.json:259]: We add `k8s` prefix to definitions, that are defined by Kubernetes. Also, the definition itself seems wrong. The `storageClassName` is an optional string, not an objet.
```suggestion
        "k8sStorageClassName": {
            "description": "The name of the StorageClass required by the claim. More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#class",
            "type": [
                "string",
                "null"
            ]
        },
```

File: deploy/helm/values.schema.json [hunk 2]
```diff
@@ -1312,6 +1319,9 @@
                 },
                 "resources": {
                     "$ref": "#/definitions/k8sResourceRequirements"
+                },
+                "storageClassName": {
+                    "$ref": "#/definitions/storageClassName"
```
Comment (pstranak-sw) [inline deploy/helm/values.schema.json:1324]: related to the other comment:
```suggestion
                    "$ref": "#/definitions/k8sStorageClassName"
```

File: deploy/helm/values.schema.json
Comment (pstranak-sw) [inline deploy/helm/values.schema.json:1]: Also add the default (undefined) value to the `values.yaml`. Not strictly necessary, but we add all default values there.
```diff
diff --git a/deploy/helm/values.yaml b/deploy/helm/values.yaml
index ef0b673..421ff19 100644
--- a/deploy/helm/values.yaml
+++ b/deploy/helm/values.yaml
@@ -690,6 +690,7 @@ swoagent:
       cpu: 100m
   nodeSelector: {}
   affinity: {}
+  storageClassName:
 
 aws_fargate:
   # Enable support for AWS EKS Fargate environment
```

**Review by pstranak-sw — CHANGES_REQUESTED** [review]

File: deploy/helm/values.schema.json
Comment (etichy) [reply deploy/helm/values.schema.json:1]: done, thanks 👍

**Review by pstranak-sw — APPROVED** [review]

**Pull request merged**